src/build123d/topology.py

```diff
@@ -8592,7 +8379,58 @@ def chamfer_2d(
         Returns:
             Wire: chamfered wire
         """
-        return Face(self).chamfer_2d(distance, distance2, vertices, edge).outer_wire()
+        reference_edge = edge
+        del edge
```

Comment:
Why do you need to del the reference towards the edge?

Reply:
Eliminated